vermouth/processors/make_bonds.py

```diff
@@ -94,7 +94,11 @@ def _bonds_from_distance(graph, nodes=None, non_edges=None, fudge=1.0):
             if subn in nodes and graph.nodes[subn].get('element') in VDW_RADII
         )
     }
-    max_dist = max(VDW_RADII.values()) * fudge
+    if idx_to_nodenum:
+        max_dist = max(VDW_RADII[graph.nodes[idx]['element']] for idx in idx_to_nodenum.values())
+    else:
+        max_dist = 0
+    max_dist *= fudge
```

Comment:
This change is also for the speed up or does that fix an error? Basically idx_to_nodenum is Flase if an element is not in VDW_radii? The previous functions is just horrible to read. So I don't really have a good grasp on what that dict actually contains.

Reply:
It's a speedup. It used to take the largest atom (Se currently) to determine the maximum possible cutoff for the pairs. Now it takes the largest atom present in the system.
And indeed. nodes that have an "unknown" element are not present in idx_to_nodenum, which maps indices in the coordinate array to node keys. The if/else guards the case where there are no atoms with known elements.

I'll add a comment.